silx/io/fabioh5.py

```diff
@@ -82,6 +83,68 @@ def supported_extensions():
     return _fabio_extensions
 
 
+_parse_numeric_value = re.compile("[-+]?(\d*)(?:\.(\d*))?(?:[eE]([-+]?\d+))?")
+"""Match integer or floating-point numbers"""
+
+
+def min_numerical_convertible_type(string):
+    """
+    Parse the string and return the minimal numerical type while fit for a
+    convertion.
+
+    :param str string: Representation of a float/integer with text
+    :raise ValueError: When the string is not a numerical value
+    :retrun: A numpy numerical type
+    """
+    if string == "":
+        raise ValueError("Not a numerical value")
+    match = _parse_numeric_value.match(string)
+    if match.end() != len(string):
+        raise ValueError("Not a numerical value")
+    number, decimal, exponent = match.groups()
+    if decimal is None and exponent is None:
+        # TODO: We could find the int type without converting the number
+        value = int(string)
+        return numpy.min_scalar_type(value).type
+    else:
+        if number is None:
+            number = 0
+        else:
+            number = len(number)
+        if decimal is None:
+            decimal = 0
+        else:
+            decimal = len(decimal)
+        if exponent is None:
+            exponent = 0
+        else:
+            exponent = int(exponent)
+
+        if exponent > 0:
+            digits = number + max(decimal, exponent)
+        elif exponent < 0:
+            exponent = -exponent
+            digits = max(number, exponent) + decimal
+        else:
+            digits = number + decimal
+
+        if digits <= 3:
+            # A float16 is accurate with about 3.311 digits
+            return numpy.float16
```

Comment:
float16 is not standard in numpy and may cause a lot of conversion error. I would start with float32